### ISSUE TITLE
Google Maps Flutter Padding Bug

### ISSUE THREAD

**Issue by stumcdonald70** [flutter/flutter#90313]
## Steps to Reproduce

1. Copy google_maps_flutter 2.0.10 Sample Usage Code`.
2. Add a Stack to the body of the Scaffold and Container with left padding and padding attributes in the GoogleMap 

class: ...  

```dart
Stack(children: [
   GoogleMap(
     padding: EdgeInsets.only(left: 100.0),
     mapType: MapType.hybrid,
     initialCameraPosition: _kGooglePlex,
     onMapCreated: (GoogleMapController controller) {
        _controller.complete(controller);
     }, 
    ),
   Container(
     width: 100.0,
     decoration: BoxDecoration(color: Colors.black54),
   ),
 ]),
  floatingActionButton: FloatingActionButton.extended(
     onPressed: _goToTheLake,
     label: Text('To the lake!'),
     icon: Icon(Icons.directions_boat),
  ),
); 
```
3. Run main.app on an Android Phone

**Expected results:** 

1. Map of Googleplex sample location with Google logo padding width from the left of 100.0 and the target to also be padded 100.0 from left
2. Click on Button "To the lake!" to animate Camera to target which should be padding from the left 100.0

**Actual results:** 

1. Map was not padded 100.0 on the initial main.app run and the target was also not padded 100.0. Had to Hot Reload app for padding to take effect and the Google logo to be in the correct position.
2. Clicked on Button, however, the zoomed-in target is still not in the correct position (padding 100.0). I click on the "To the lake!" button for the 2nd time and the target moved to the expected position (padding of 100.0)

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.1 20C69 darwin-x64, locale en-US)
    • Flutter version 2.5.1 at /Users/quidapps/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (8 hours ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc1)
    • Android SDK at /Users/quidapps/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • SM A125U (mobile) • R58R35WLBXT • android-arm64  • Android 10 (API 29)
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

</details>

**Comment by maheshmnj** [comment]
Hi @stumcdonald70, Thanks for filing the issue. I can replicate the issue on Android and works fine on IOS. When the map loads for the first time the padding is not applied but on hot reload the padding takes effect. Verified on the latest `stable 2.5.1` with `google_maps 2.0.10`


### IOS

Uploading Screen Recording 2021-09-20 at 6.06.18 PM.mov…


### Android
Uploading Screen Recording 2021-09-20 at 6.00.29 PM.mov…


<details>
<summary>code sample</summary>

```dart
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

class MapSample extends StatefulWidget {
  @override
  State<MapSample> createState() => MapSampleState();
}

class MapSampleState extends State<MapSample> {
  Completer<GoogleMapController> _controller = Completer();

  static final CameraPosition _kGooglePlex = CameraPosition(
    target: LatLng(37.42796133580664, -122.085749655962),
    zoom: 14.4746,
  );

  static final CameraPosition _kLake = CameraPosition(
      bearing: 192.8334901395799,
      target: LatLng(37.43296265331129, -122.08832357078792),
      tilt: 59.440717697143555,
      zoom: 19.151926040649414);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(children: [
        GoogleMap(
          padding: EdgeInsets.only(left: 100.0),
          mapType: MapType.hybrid,
          initialCameraPosition: _kGooglePlex,
          onMapCreated: (GoogleMapController controller) {
            _controller.complete(controller);
          },
        ),
        Container(
          width: 100.0,
          decoration: BoxDecoration(color: Colors.black54),
        ),
      ]),
      floatingActionButton: FloatingActionButton.extended(
        onPressed: _goToTheLake,
        label: Text('To the lake!'),
        icon: Icon(Icons.directions_boat),
      ),
    );
  }

  Future<void> _goToTheLake() async {
    final GoogleMapController controller = await _controller.future;
    controller.animateCamera(CameraUpdate.newCameraPosition(_kLake));
  }
}

void main() {
  runApp(MaterialApp(home: MapSample()));
}


```
</details>


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.6 20G165 darwin-arm, locale en-GB)
    • Flutter version 2.5.1 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (2 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.59.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • iPad Pro (11-inch) (3rd generation) (mobile) • 4367E72A-10EB-46BA-9E12-D89A8F0DA157 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)                              • macos                                • darwin-arm64   • macOS 11.6 20G165 darwin-arm
    • Chrome (web)                                 • chrome                               • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

```
[✓] Flutter (Channel master, 2.6.0-6.0.pre.147, on macOS 11.6 20G165 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-6.0.pre.147 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 83dfb2237a (2 days ago), 2021-09-17 23:48:04 -0400
    • Engine revision 31792e0340
    • Dart version 2.15.0 (build 2.15.0-120.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.59.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile) • d9dac45d                             • android-arm64  • Android 11 (API
      30)
    • iPhone 12 Pro (mobile) • B622CBB7-F906-4FA7-8F49-FACEAAC905AB • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 11.6 20G165
      darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome
      93.0.4577.82

• No issues found!
```

</details>

**Comment by Mayb3Nots** [comment]
this should be closed in favor of #106738

**Comment by stuartmorgan** [comment]
Closing as a duplicate of #43721.
